### PR TITLE
Feature: Initialization

### PR DESCRIPTION
**Pull request by aboubakar14**

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[packages]
+fastapi = "*"
+uvicorn = {extras = ["standard"], version = "*"}
+requests = "*"
+black = "*"
```
Comment (sohzaz) [inline Pipfile]: Nit: Shouldn't black be a dev package?

File: .github/workflows/CICD-check.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    - name: Install dependencies
+      run: python -m pip install --upgrade pip
+    - name: Setup virtualenv
+      run: |
+       pip install pipenv
+       pipenv install
+       pipenv install --dev
+    - name: Check lint with Black
+      run: |
+       pipenv run pip install black
```
Comment (sohzaz) [inline .github/workflows/CICD-check.yml]: Isn't black already installed by being present in the Pipfile?

File: sonar-project.properties [hunk 3]
```diff
@@ -0,0 +1,8 @@
+sonar.projectKey=ditrit_leto-modelizer-ia-api
+sonar.organization=ditrit
+sonar.projectName=leto-modelizer-ia-api
+sonar.host.url=https://sonarcloud.io
+sonar.coverage.exclusions=**/tests/**
+sonar.sources=./src
+sonar.exclusions=node_modules/**
```
Comment (sohzaz) [inline sonar-project.properties]: Sources and exclusions will probably need to be adjusted to match the project structure later.

**Review by sohzaz — DISMISSED** [review]
LGTM, maybe a couple of nitpicks to look at but non-blocking.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,37 @@
 # leto-modelizer-ia-api
+
+## Description
 Project to manage api to request ia.
+
+### Requirements
```
Comment (Zorin95670) [inline README.md]: move all in contributing

File: .github/workflows/CICD-check.yml [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    - name: Setup virtualenv
+      run: |
+       pip install pipenv
+       pipenv install
+       pipenv install --dev
+    - name: Check lint with Black
+      run: pipenv run black . --check
+
+
+  check-dependencies:
```
Comment (Zorin95670) [inline .github/workflows/CICD-check.yml]: Please send a comment on github like over projects

File: changelog.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html)
+
+## [Unreleased]
```
Comment (Zorin95670) [inline changelog.md:8]: remove all sub section except Added and add one line like `- Setup project`

Comment (aboubakar14) [reply changelog.md:8]: done

File: .github/workflows/CICD-check.yml (hunk 5 above)
Comment (aboubakar14) [reply .github/workflows/CICD-check.yml]: done

File: README.md (hunk 4 above)
Comment (aboubakar14) [reply README.md]: done

File: sonar-project.properties [hunk 7]
```diff
@@ -0,0 +1,7 @@
+sonar.projectKey=ditrit_leto-modelizer-ia-api
+sonar.organization=ditrit
+sonar.projectName=leto-modelizer-ia-api
+sonar.host.url=https://sonarcloud.io
+sonar.coverage.exclusions=**/tests/**
+sonar.sources=./src
+sonar.python.coverage.reportPaths=coverage.xml
```
Comment (justinehell) [inline sonar-project.properties]: missing end line

File: .github/workflows/CI.yml [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+    - name: Unit tests with pytest
+      run: |
+       pipenv run pip install pytest pytest-cov
+       pipenv run pytest --cov=. --cov-report=xml  
+    - name: SonarCloud Scan
+      if: always()
+      uses: SonarSource/sonarcloud-github-action@master
+      env: 
+        SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (justinehell) [inline .github/workflows/CI.yml]: missing end line

File: .github/workflows/clean.yml [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+        with:
+          token: ${{ github.token }}
+          repository: ${{ github.repository }}
+          retain_days: 7
+          keep_minimum_runs: 0
+          delete_run_by_conclusion_pattern: All
+      - name: Purge cache
+        uses: MyAlbum/purge-cache@v1
+        with:
+          max-age: 604800 # Cache max 7 days since last use
```
Comment (justinehell) [inline .github/workflows/clean.yml]: missing end line

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,86 @@
+# leto-modelizer-ia-api
+
+## Description
+This file describe how to contribute and develop to this open source projet.
```
Comment (justinehell) [inline CONTRIBUTING.md]: This file describe**s** how to contribute to and develop for this open-source project.

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+
+    Install pipenv:  ``` pip install pipenv ```
+
+    Create virtual environment and install dependencies from your repo: ``` pipenv install ```
+
+    Install dev dependencies: ``` pipenv install --dev ```
+
+    Activate virtual environment: ``` pipenv shell ```
+
+    To deactivate your virtual environment: ``` deactivate ```
```
Comment (justinehell) [inline CONTRIBUTING.md:18]: only `deactivate` ?

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+
+
```
Comment (justinehell) [inline CONTRIBUTING.md]: remove one empty line

**Review by justinehell — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 11 above)
Comment (Zorin95670) [inline CONTRIBUTING.md:19]: ha ha ha same

Comment (Zorin95670) [reply CONTRIBUTING.md:19]: I say it, it's not clear at all ;)

**Review by Zorin95670 — APPROVED** [review]

**Pull request merged**